core/calculation_debug.go

```diff
@@ -0,0 +1,418 @@
+	}
+}
+
+func WriteBeta3Info(ctx *Context, rewardAddress common.Address, rewardRep uint64,
+	delegationInfo *DelegateData, period uint64, endBlock uint64) {
+	if !NeedToUpdateCalcDebugResult(ctx) {
+		return
+	}
+	for _, address := range ctx.calcDebug.conf.Addresses {
+		writeBeta3Info(ctx, address, rewardAddress, rewardRep, delegationInfo, period, endBlock)
+	}
+}
+
+func writeBeta3Info(ctx *Context, debugAddress *common.Address, rewardAddress common.Address, rewardRep uint64,
+	delegationInfo *DelegateData, period uint64, endBlock uint64) {
+	if rewardAddress.Equal(debugAddress) {
+		calcResult := GetCalcResult(ctx, *debugAddress)
+		reward := getReward(calcResult, endBlock)
+		iScore := rewardRep * period * delegationInfo.Delegate.Uint64() / rewardDivider
+		dgInfo := &DelegationInfo{BlockHeight: endBlock, Address: delegationInfo.Address,
+			Amount: delegationInfo.Delegate.Uint64(), IScore: iScore}
+		dgInfo.Formula = fmt.Sprintf("%s * %d * %d / %d", common.NewHexIntFromUint64(rewardRep).String(), period,
+			dgInfo.Amount, rewardDivider)
+		reward.Beta3.DelegationInfo = append(reward.Beta3.DelegationInfo, dgInfo)
+		iScoreInHex := *common.NewHexIntFromUint64(iScore)
+		reward.Beta3.Beta3IScore.Add(&reward.Beta3.Beta3IScore.Int, &common.NewHexIntFromUint64(iScore).Int)
+		calcResult.TotalIScore.Add(&calcResult.TotalIScore.Int, &iScoreInHex.Int)
+	}
+}
+
+func GetCalcResult(ctx *Context, address common.Address) *CalcResult {
+	calcResult := getCalcResult(ctx, address)
+	if calcResult != nil {
+		return calcResult
+	}
+	InitCalcResult(ctx, address)
+	calcResult = getCalcResult(ctx, address)
+	return calcResult
+}
+
+func getCalcResult(ctx *Context, address common.Address) *CalcResult {
+	for _, calcResult := range ctx.calcDebug.result.Results {
+		if calcResult.Address.Equal(&address) {
+			return calcResult
+		}
+	}
+	return nil
+}
+
+func getReward(calcResult *CalcResult, blockHeight uint64) *Reward {
+	var r *Reward
+	rewardLength := len(calcResult.Rewards)
+	for i := rewardLength - 1; i >= 0; i-- {
+		reward := calcResult.Rewards[i]
+		if reward.BlockHeight < blockHeight {
+			r = reward
+		}
+	}
+	return r
+}
+
+func InitCalcResult(ctx *Context, address common.Address) {
+	result := NewCalcResult(&address)
+	initialIScore := *common.NewHexInt(0)
+	qDB := ctx.DB.getQueryDB(address)
+	bucket, _ := qDB.GetBucket(db.PrefixIScore)
+	bs, _ := bucket.Get(address.Bytes())
+	if bs != nil {
+		ia, _ := NewIScoreAccountFromBytes(bs)
+		initialIScore = ia.IScore
+	}
+	result.InitialIScore = initialIScore
+	result.TotalIScore = initialIScore
+
+	rewards := make([]*Reward, 0)
+	for _, gv := range ctx.GV {
+		reward := NewReward(gv.BlockHeight)
+		rewards = append(rewards, reward)
+	}
+	result.Rewards = rewards
+	ctx.calcDebug.result.Results = append(ctx.calcDebug.result.Results, result)
+}
+
+func WriteCalcDebugResult(ctx *Context) {
+	calcDebugDB := db.Open(ctx.DB.info.DBRoot, string(db.GoLevelDBBackend), "calculation_debug")
```

Comment:
How about using ctx.calcDebug.conf.Output as the DB path?

Reply:
ctx.calcDebug.conf.Output was previously required to output the results to a file, but was deleted as it was no longer needed while storing the results in the DB.